pkg/discovery/probe/pod_probe.go

```diff
@@ -208,30 +208,40 @@ func (podProbe *PodProbe) getPodResourceStat(pod *api.Pod, podContainers map[str
 	if containers, ok := podContainers[podNameWithNamespace]; ok {
 		for _, container := range containers {
 			containerStats := container.Stats
+			// We need at least two data points.
 			if len(containerStats) < 2 {
 				//TODO, maybe a warning is enough?
 				glog.Warningf("Not enough data for %s. Skip.", podNameWithNamespace)
 				continue
-				// return nil, fmt.Errorf("Not enough data for %s", podNameWithNamespace)
 			}
+			// Get the average of all the available data.
+			dataSampleNumber := len(containerStats)
+			glog.V(4).Infof("data sample number is %d", dataSampleNumber)
+
 			currentStat := containerStats[len(containerStats)-1]
-			prevStat := containerStats[len(containerStats)-2]
+			prevStat := containerStats[len(containerStats)-dataSampleNumber]
```

Comment:
"len(containerStats) - dataSampleNumber"  is not useful, because dataSampleNumber is always equal to len(containerStats).   Should define dataSampleNumber to be other number.